src/TimeoutMigrationTool/RavenDB/RavenDBTimeoutStorage.cs

```diff
@@ -151,30 +150,33 @@ public async Task Complete()
             toolState.Status = MigrationStatus.Completed;
 
             var ravenToolState = RavenToolState.FromToolState(toolState);
-            var archivedToolStateId = $"{RavenConstants.ArchivedToolStateIdPrefix}{toolState.Endpoint.EndpointName}-{DateTime.Now.ToShortTimeString()}";
-            await ravenAdapter.ArchiveDocument(archivedToolStateId, ravenToolState);
+            await ravenAdapter.ArchiveDocument(GetArchivedToolStateId(toolState), ravenToolState);
         }
 
-        internal async Task CleanupExistingBatchesAndResetTimeouts(List<BatchInfo> batchesToRemove,
-            List<BatchInfo> batchesForWhichToResetTimeouts)
+        internal async Task CleanupExistingBatchesAndResetTimeouts(List<BatchInfo> batchesToRemove, List<BatchInfo> batchesForWhichToResetTimeouts)
         {
             foreach (var batch in batchesToRemove)
+            {
                 if (batchesForWhichToResetTimeouts.Any(b => b.Number == batch.Number))
+                {
                     await ravenAdapter.DeleteBatchAndUpdateTimeouts(batch);
+                }
                 else
+                {
                     await ravenAdapter.DeleteDocument($"{RavenConstants.BatchPrefix}/{batch.Number}");
+                }
+            }
         }
 
-        internal async Task RemoveToolState()
-        {
-            await ravenAdapter.DeleteDocument(RavenConstants.ToolStateId);
-        }
-
-
         async Task StoreToolState(ToolState toolState)
         {
             var ravenToolState = RavenToolState.FromToolState(toolState);
             await ravenAdapter.UpdateDocument(RavenConstants.ToolStateId, ravenToolState);
         }
+
+        string GetArchivedToolStateId(ToolState toolState)
+        {
+            return $"{RavenConstants.ArchivedToolStateIdPrefix}{toolState.Endpoint.EndpointName}-{DateTime.Now:yyyy-MM-dd hh-mm-ss}";
```

Comment:
Are there any limits on Document ID length in Raven?

Reply:
There are following limitations for document IDs:

The identifier length limit is 512 bytes (in UTF8)
The identifier cannot contain \ character
You can't store identifier ending with / or | characters (reserved for Server-Side ID and Identity generation strategies)